posthog/api/test/test_survey.py

```diff
@@ -1742,6 +1742,75 @@ def test_custom_question_ids_accepted_when_creating_survey(self):
         self.assertEqual(response_data["questions"][0]["id"], custom_id_1)
         self.assertEqual(response_data["questions"][1]["id"], custom_id_2)
 
+    def test_survey_search_functionality(self):
+        # Create multiple surveys
+        self.client.post(
+            f"/api/projects/{self.team.id}/surveys/",
+            data={
+                "name": "Product Feedback Survey",
+                "description": "Get feedback on our product",
+                "type": "popover",
+                "questions": [{"type": "open", "question": "What do you think?"}],
+                "targeting_flag_filters": None,
+            },
+            format="json",
+        ).json()
+
+        self.client.post(
+            f"/api/projects/{self.team.id}/surveys/",
+            data={
+                "name": "NPS Survey 2024",
+                "description": "Annual NPS survey",
+                "type": "popover",
+                "questions": [{"type": "open", "question": "Would you recommend us?"}],
+                "targeting_flag_filters": None,
+            },
+            format="json",
+        ).json()
+
+        # Test search by name
+        response = self.client.get(f"/api/projects/{self.team.id}/surveys/?search=NPS")
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        data = response.json()
+        self.assertEqual(len(data["results"]), 1)
+        self.assertEqual(data["results"][0]["name"], "NPS Survey 2024")
+
+        # Test search by description
+        response = self.client.get(f"/api/projects/{self.team.id}/surveys/?search=product")
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        data = response.json()
+        self.assertEqual(len(data["results"]), 1)
+        self.assertEqual(data["results"][0]["name"], "Product Feedback Survey")
+
+        # Test search with no results
+        response = self.client.get(f"/api/projects/{self.team.id}/surveys/?search=nonexistent")
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        data = response.json()
+        self.assertEqual(len(data["results"]), 0)
+
+        # Test search with pagination
+        # Create more surveys to test pagination
```

Comment:
I think each of these blocks should be its own test, so when it fails, we know what to look at